src/shelly_temp_sensor_ow.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) Shelly-HomeKit Contributors
+ * All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "shelly_temp_sensor_ow.hpp"
+
+#ifdef MGOS_HAVE_ONEWIRE
+
+#include <cmath>
+
+#include <mgos_onewire.h>
+
+/* Model IDs */
+#define DS18S20MODEL 0x10 /* also DS1820 */
+#define DS18B20MODEL 0x28
+#define DS1822MODEL 0x22
+#define DS1825MODEL 0x3B
+
+/* Scratchpad size */
+#define SCRATCHPAD_SIZE 9
+
+/* Scratchpad locations */
+#define CONFIGURATION 4
+#define SCRATCHPAD_CRC 8
+
+/* Device resolution */
+#define TEMP_9_BIT 0x1F  /* 9 bit */
+#define TEMP_10_BIT 0x3F /* 10 bit */
+#define TEMP_11_BIT 0x5F /* 11 bit */
+#define TEMP_12_BIT 0x7F /* 12 bit */
+
+#define CONVERT_T 0x44
+#define READ_SCRATCHPAD 0xBE
+#define READ_POWER_SUPPLY 0xB4
+
+namespace shelly {
+
+Onewire::Onewire(int pin_in, int pin_out) {
+  ow_ = mgos_onewire_create_separate_io(pin_in, pin_out);
+}
+
+Onewire::~Onewire() {
+  mgos_onewire_close(ow_);
+}
+
+struct mgos_onewire *Onewire::Get() {
+  return ow_;
+}
+
+void Onewire::DiscoverAll(int num_sensors_max,
```

Comment:
i have a question about discovery: does onewire guarantee the order of discovered sensors? i.e., is the set of the same sensors guaranteed to be mapped to se{1,2,3} in the same order every time?

Reply:
yes it does, see https://forum.arduino.cc/t/order-of-detecting-1-wire-devices/6395/4